Consumer/src/main/java/com/finalboss/useCases/EventHandler.java

```diff
@@ -0,0 +1,101 @@
+     * If the Event exists then verify if it has a Market with the id received in the email.
+     * If not, add that market to the list of markets of that Event.
+     * If already exists then you can ignore that email.
+     */
+    private void addYellowEventToRepository(MarketUpdate update) throws JsonProcessingException {
+        // Check the Yellow Events to verify if an Event with the ID received in event.id already exists.
+        YellowEvent yellowEventUpdate = yellowEventMapper.buildYellowEvent(update);
+
+        if (yellowEventRepositoryI.existsById(yellowEventUpdate.id())) {
+            // If the Event exists then verify if it has a Market with the id received in the email.
+            Optional<YellowEvent> event = yellowEventRepositoryI.findById(yellowEventUpdate.id());
+            List<Market> markets = event.get().markets();
+
+            boolean hasMarket = false;
+            for (Market market : markets) {
+                //If already exists then you can ignore that email.
+                if (market.id().equals(update.id())) {
+                    hasMarket = true;
+                    break;
+                }
+            }
+            //If not, add that market to the list of markets of that Event.
+            if (!hasMarket) {
+                event.get().markets().add(yellowEventUpdate.markets().getFirst());
+                yellowEventRepositoryI.save(event.get());
+            }
+        }
+        // If the Event does not exist then it must be added with the market received.
+        yellowEventRepositoryI.save(yellowEventUpdate);
+    }
+
+    /**
+     * Check the Yellow Events to verify if an Event with the ID received in event.id already exists.
+     * If it exists then verify if it has a Market with the id received in the email.
+     * If it already exists then you should update its name and selections with the ones received in the email.
+     * If it doesn't exist you can ignore that email.
+     */
+    private void modifyYellowEventInRepository(MarketUpdate update) throws JsonProcessingException {
+        //Check the Yellow Events to verify if an Event with the ID received in event.id already exists.
+        YellowEvent yellowEventUpdate = yellowEventMapper.buildYellowEvent(update);
+        if (yellowEventRepositoryI.existsById(yellowEventUpdate.id())) {
+            //If it exists then verify if it has a Market with the id received in the email.
+            Optional<YellowEvent> event = yellowEventRepositoryI.findById(yellowEventUpdate.id());
+            List<Market> markets = event.get().markets();
+
+            for (Market market : markets) {
+                //If it already exists then you should update its name and selections with the ones received in the email.
+                if (market.id().equals(update.id())) {
+                    market = yellowEventUpdate.markets().getFirst(); //??
+                    yellowEventRepositoryI.save(event.get());
+                }
```

Comment:
If you reason about what you are trying to do here, it goes something like this:
1. check if there is a market with a given id
2. if true, get the first (and only) market of the update
3. get the market that was matched, and update its name and selections with the ones on the update
4. save the event that you got from your db

You are doing everything, except for point 3.
Try to implement it! :)

Reply:
I think its done, but it doesn't feel right